prow/resultstore/payload.go

```diff
@@ -151,37 +160,68 @@ func (p *Payload) invocationProperties() []*resultstore.Property {
 			Value: p.Job.Status.URL,
 		},
 	}
-	if p.Job.Spec.Refs != nil {
-		ps = append(ps, &resultstore.Property{
-			Key:   "Repo",
-			Value: p.Job.Spec.Refs.RepoLink,
-		})
-	}
-	if p.Started != nil {
-		ps = append(ps, &resultstore.Property{
-			Key:   "Commit",
-			Value: p.Started.RepoCommit,
-		})
-		for _, branch := range p.Started.Repos {
-			ps = append(ps, &resultstore.Property{
-				Key:   "Branch",
-				Value: branch,
-			})
-		}
+	ps = append(ps, p.podSpecProperties()...)
+	ps = append(ps, p.startedProperties()...)
+	return ps
+}
+
+func (p *Payload) podSpecProperties() []*resultstore.Property {
+	var ps []*resultstore.Property
+	if p.Job.Spec.PodSpec == nil {
+		return ps
 	}
-	if p.Job.Spec.PodSpec != nil {
-		for _, c := range p.Job.Spec.PodSpec.Containers {
-			for _, e := range c.Env {
+	seenEnv := map[string]bool{}
+	for _, c := range p.Job.Spec.PodSpec.Containers {
+		for _, e := range c.Env {
+			v := e.Name + "=" + e.Value
+			if !seenEnv[v] {
+				seenEnv[v] = true
 				ps = append(ps, &resultstore.Property{
 					Key:   "Env",
-					Value: e.Name + "=" + e.Value,
+					Value: v,
 				})
 			}
 		}
 	}
 	return ps
 }
 
+func (p *Payload) startedProperties() []*resultstore.Property {
```

Comment:
Nit: A comment explaining what these properties are and where they come from would be helpful. Is it coming from `started.json`?

Reply:
Refactored to improve clarity of data flow.